src/MassTransit/DependencyInjection/DependencyInjection/ScopedMediator.cs

```diff
@@ -0,0 +1,248 @@
+namespace MassTransit.DependencyInjection
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Clients;
+    using Context;
+    using Mediator;
+    using Transports;
+
+
+    public class ScopedMediator :
+        SendEndpointProxy,
+        IScopedMediator
+    {
+        readonly ClientFactory _clientFactory;
+        readonly IMediator _mediator;
+        readonly IServiceProvider _provider;
+
+        public ScopedMediator(IMediator mediator, IServiceProvider provider)
+            : base(mediator)
+        {
+            _mediator = mediator;
+            _provider = provider;
+            _clientFactory = new ClientFactory(new ScopedClientFactoryContext(_mediator, _provider));
```

Comment:
Look at how [this](https://github.com/MassTransit/MassTransit/blob/68eb9052c33d6a928c71fcde8426b0f787f84467/src/MassTransit/DependencyInjection/DependencyInjection/BusScopedBusContext.cs#L41) does it, lazy initialized, and using ScopedClientFactory.